src/sync.js

```diff
@@ -3,6 +3,31 @@ import PQueue from 'p-queue'
 import Path from 'path'
 import { EventEmitter } from 'events'
 
+/**
+ * @description
+ * Syncs an append-only, conflict-free replicated data type (CRDT) log between multiple peers.
+ *
+ * Sync is responsible for synchronizing heads between multiple peers.
+ * - When Sync is started, heads are sent by peers sharing the same log id.
+ * - Heads are received by each peer from another peer with the same log id.
+ * - There is no guarantee that heads will be received in the order they were generated on the sending peer.
+ * - There is no guarantee that all heads will be received.
+ * - There is no guarantee that heads will reflect the latest heads on the remote peer.
+ * - Calling Sync.add is neither synchronous nor deterministic. Calling Sync.add with async/await only guarantees that the message is published. It does not guarantee when the message will be received or even that the message is recieved at all.
+ * - Determining whether a peer has received a message can be handled by registering a callback using onSynced. However, there is no guarantee that every log entry will be received and onSynced can not be used to determine whether every head is received.
+ * - The only guarantee that Sync provides is that heads will eventually reach consistency between all peers.
+ */
+
+/**
+ * Creates a Sync instance for sychronizing logs between multiple peers.
+ * @param {Object} options One or more options for configurating Sync.
```

Comment:
I think the params here are not options (as in optionals) but rather required params.